### PR TITLE
Consider other conditions while listing templates with id

### PR DESCRIPTION
**Pull request by ravening**
## Description

Many of the searching conditions are not considered when
template id is passed but they are considered only when
template id is not passed. Move other conditions out of else
part so that they will be considered when id is passed.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

Without changes

```
(local) mgt01 > list templates templatefilter=all filter=id,hypervisor, id="45be18c7-c0fd-11ea-ba74-067606003392"
{
  "count": 1,
  "template": [
    {
      "hypervisor": "KVM",
      "id": "45be18c7-c0fd-11ea-ba74-067606003392"
    }
  ]
}
(local) mgt01 >
(local) mgt01 >
(local) mgt01 > list templates templatefilter=all filter=id,hypervisor, id="45be18c7-c0fd-11ea-ba74-067606003392" hypervisor=XenServer
{
  "count": 1,
  "template": [
    {
      "hypervisor": "KVM",
      "id": "45be18c7-c0fd-11ea-ba74-067606003392"
    }
  ]
}
```

With changes
```
(local) mgt01 > list templates templatefilter=all filter=id,hypervisor id="d9411172-3832-4fc9-a771-39514e0a9faa" hypervisor=KVM
{
  "count": 1,
  "template": [
    {
      "hypervisor": "KVM",
      "id": "d9411172-3832-4fc9-a771-39514e0a9faa"
    }
  ]
}
(local) mgt01 > list templates templatefilter=all filter=id,hypervisor id="d9411172-3832-4fc9-a771-39514e0a9faa" hypervisor=XenServer
(local) mgt01 >
```

### CODE REVIEW

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java [hunk 1]
```diff
@@ -3434,11 +3434,6 @@ else if (!template.isPublicTemplate() && caller.getType() != Account.ACCOUNT_TYP
                 domain = _domainDao.findById(Domain.ROOT_DOMAIN);
             }
 
-            // List<HypervisorType> hypers = null;
-            // if (!isIso) {
-            // hypers = _resourceMgr.listAvailHypervisorInZone(null, null);
-            // }
-
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:3441]: 👍

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java [hunk 2]
```diff
@@ -3486,17 +3481,6 @@ else if (!template.isPublicTemplate() && caller.getType() != Account.ACCOUNT_TYP
...
-                // add hypervisor criteria for template case
-                if (hypers != null && !hypers.isEmpty()) {
-                    String[] relatedHypers = new String[hypers.size()];
-                    for (int i = 0; i < hypers.size(); i++) {
-                        relatedHypers[i] = hypers.get(i).toString();
-                    }
-                    sc.addAnd("hypervisorType", SearchCriteria.Op.IN, relatedHypers);
-                }
-            }
-
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:3499]: can you make it a method and put the call ...

**Review by DaanHoogland — APPROVED** [review]
more diff than changes, code looks good

File: server/src/main/java/com/cloud/api/query/QueryManagerImpl.java (hunk 2 above)
Comment (ravening) [reply server/src/main/java/com/cloud/api/query/QueryManagerImpl.java:3499]: @DaanHoogland  done

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@rhtyd simple fix for API usage quirk, can you review? cc @PaulAngus

**Review by weizhouapache — APPROVED** [review]

**Pull request merged**